pkg/shared/utils/sync/distributed_lock_mgr_test.go

```diff
@@ -0,0 +1,203 @@
+package sync
+
+import (
+	"fmt"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/shared/utils/arrays"
+	"github.com/onsi/gomega"
+	"gorm.io/gorm"
+	"sync"
+	"sync/atomic"
+	"testing"
+	"time"
+)
+
+var _ DistributedLock = &__distributedLockMock{}
+
+// this map is used to simulate database locks
+var mutexMap map[string]*sync.Mutex
+
+// this lock is used to sync access to the mutexMap
+var mu sync.Mutex
+
+func getMutex(lockID string) *sync.Mutex {
+	mu.Lock()
+	defer mu.Unlock()
+	if mutexMap == nil {
+		mutexMap = map[string]*sync.Mutex{}
+	}
+
+	testLock, ok := mutexMap[lockID]
+	if !ok {
+		testLock = &sync.Mutex{}
+		mutexMap[lockID] = testLock
+	}
+	return testLock
+}
+
+type __distributedLockMock struct {
```

Comment:
Why is this created and not the `distributedLockMock`?

Reply:
Because the `distributedLockMock` is based on a mocked database, while this one based on mutex.
It made creating unit tests for the `distributed_lock_mgr` easier without the need to mock the database.

Reply:
Could we name it inMemoryDistributedLock or something like that then?